openquake/mbt/tools/fault_modeler/fault_modeling_utils.py

```diff
@@ -3112,11 +3114,13 @@ def calc_youngs_coppersmith_mfd_from_fault_params(
 
     moment_rate = (seismic_slip_rate * 1e-3) * (fault_area * 1e6) * rigidity
 
-    mfd = hz.mfd.YoungsCoppersmith1985MFD.from_total_moment_rate(m_cli,
+    mfd = hz.mfd.YoungsCoppersmith1985MFD.from_total_moment_rate(m_min,
                                                                  b_value,
                                                                  m_char,
                                                                  moment_rate,
                                                                  bin_width)
+    
+    mfd.min_mag = m_cli
```

Comment:
I see what you're doing here, and I think it's an efficient way of solving the problem (although changing the attribute of an existing instance of a class that has a lot of computation done during instantiation is *generally* a bad idea, so having a comment to explain what you are doing and why is necessary).  However, I think that it can lead to problems if the `m_cli` parameter doesn't correspond to the edge of one of the magnitude bins that has already been defined.  The problems will be that the new rounding of the MFD doesn't match the older rounding, so that the moment balancing (and therefore occurrence rates) may not be correct.

In Julio's fix for the GR, he makes the final MFD out of the bins that are output initially, using only bins that are greater than or equal to `m_cli`.  This is one option, and we could return an EvenlyDiscretizedMFD from the results.  This is an easy fix, and has the appeal that the API of the two functions is consistent, which can be helpful.

Another option is to adjust `m_cli` (or to define `m_cli_rounded` or something, instead of re-defining a parameter) so that it falls on the bin edge.  Please see the [how this rounding is done in the YC1985 definition](https://github.com/gem/oq-engine/blob/80d8a15f58f0f887f556bfb42085b1e55d8c2e36/openquake/hazardlib/mfd/youngs_coppersmith_1985.py#L339).  This returns the bin centers, so you would have to take the first bin that is output and then subtract half a bin width.  Alternately you can just add bin widths to `m_min` as already defined, until you reach the constraints

`m_min` + n * `bin_width` < `m_cli`

and

`m_min` + (n+1) * bin_width` > `m_cli`

and then use set `mfd.min_mag = m_min + n * bin_width`.

Reply:
I've added a similar correction to what is used by the GR, returning an EvenlyDiscretizedMFD for the results.